src/coreclr/src/vm/peimage.cpp

```diff
@@ -367,7 +367,9 @@ BOOL PEImage::CompareImage(UPTR u1, UPTR u2)
     EX_TRY
     {
         SString path(SString::Literal, pLocator->m_pPath);
-        if (PathEquals(path, pImage->GetPath()))
+        BOOL lookInBundle = pLocator->m_bLookInBundle;
+        if (PathEquals(path, pImage->GetPath()) &&
+            (lookInBundle == pImage->IsInBundle()))
```

Comment:
Comparing BOOL this way is dangerous. `BOOL` is `int` and any non-zero means TRUE. It means that e.g. one side can be 1, the other side can be 42, and the comparison will return false even though it should return true.

More robust way to compare BOOLs is: `!isInBundle == !pImage->IsInBundle()`. `!` operator will normalize the value to 0/1. You can fine it used in number of places, for example look for `_ASSERTE(!fContainsGenericVariables == !pMT->ContainsGenericVariables());` in `generics.cpp`.

Reply:
I hadn't noticed this. Thanks for the feedback.